packages/trader/src/AppV2/Containers/Trade/trade-types.tsx

```diff
@@ -1,28 +1,230 @@
 
-const TemporaryTradeTypes = ({ contract_type, onTradeTypeSelect, trade_types }: TTemporaryTradeTypesProps) => {
+const TradeTypes = ({ contract_type, onTradeTypeSelect, trade_types }: TTradeTypesProps) => {
+    const [is_open, setIsOpen] = React.useState<boolean>(false);
+    const [is_editing, setIsEditing] = React.useState<boolean>(false);
+
+    const { onMount, onUnmount } = useTraderStore();
+
+    type TItem = {
+        id: string;
+        title: string;
+        icon?: React.ReactNode;
+    };
+
+    type TResultItem = {
+        id: string;
+        title?: string;
+        button_title?: string;
+        onButtonClick?: () => void;
+        items: TItem[];
+    };
+
+    const createArrayFromCategories = (data: any): TItem[] => {
+        const result: TItem[] = [];
+
+        data.forEach((category: { value: string; text: string }) => {
+            result.push({
+                id: category.value,
+                title: category.text,
+            });
+        });
+
+        return result;
+    };
+
+    const saved_other_trade_types = localStorage.getItem('other_trade_types');
+    const saved_pinned_trade_types = localStorage.getItem('pinned_trade_types');
+
+    const [other_trade_types, setOtherTradeTypes] = React.useState<TResultItem[]>(
+        saved_other_trade_types ? JSON.parse(saved_other_trade_types) : []
+    );
+    const [pinned_trade_types, setPinnedTradeTypes] = React.useState<TResultItem[]>(
+        saved_pinned_trade_types ? JSON.parse(saved_pinned_trade_types) : []
+    );
+
+    const handleCloseTradeTypes = () => {
+        setIsOpen(false);
+        setIsEditing(false);
+    };
+
+    const handleAddPinnedClick = (item: TItem) => {
+        setOtherTradeTypes(prev_categories =>
+            prev_categories.map(category => ({
+                ...category,
+                items: category.items.filter(i => i.id !== item.id).sort((a, b) => a.title?.localeCompare(b.title)),
+            }))
+        );
+        setPinnedTradeTypes(prev_pinned => {
+            const updated_pinned = [...prev_pinned];
+            const pinned_category = updated_pinned.find(cat => cat.id === 'pinned');
+
+            if (pinned_category) {
+                pinned_category.items.push(item);
+            } else {
+                updated_pinned.push({
+                    id: 'pinned',
+                    title: localize('Pinned'),
+                    items: [item],
+                });
+            }
+
+            return updated_pinned;
+        });
+    };
+
+    const handleRemovePinnedClick = (item: TItem) => {
+        setPinnedTradeTypes(prev_categories =>
+            prev_categories.map(category => ({
+                ...category,
+                items: category.items.filter(i => i.id !== item.id),
+            }))
+        );
+        setOtherTradeTypes(prev_others => {
+            const updated_others = [...prev_others];
+            const other_category = updated_others.find(cat => cat.id === 'other');
```

Comment:
<img width="323" alt="Screenshot 2024-08-05 at 12 48 57 PM" src="https://github.com/user-attachments/assets/3a0f7a6a-0863-4005-ad14-e3c8a5fb7b36">

Reply:
Yeah, I'm not a dog person 😛